### PR TITLE
#166240951 Users should be able to Login via  Social Media

### PR DESCRIPTION
**Pull request by EmyRukundo**
**What does this PR do?**

Users should be able to Login via Social Media

 **Description of Task to be completed?**

Users should be able to Login via Google Facebook or Twitter

**How should this be manually tested?**

 **Any background context you want to provide?**

 **What are the relevant pivotal tracker stories?**

#166240951

**Screenshots (if appropriate)**

GIF
ezgif com-video-to-gif

**Questions:**

### CODE REVIEW

**Review by rukundoeric — APPROVED** [review]

File: __tests__/actions/socialLogin.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
+/* eslint-disable prefer-promise-reject-errors */
```
Comment (eliemugenzi) [inline __tests__/actions/socialLogin.js]: Great work @EmyRukundo, In order to follow naming conventions, each test file should be like this: `socialLogin.test.js`

File: src/assets/scss/layouts/_forms.scss [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+}
+#col-login{
+input[type=text], input[type=password], input[type=submit]{
+    width:92.3%;
+    height: 40px;   
+    }
+} 
+
+
+@media(max-width:768px){
```
Comment (eliemugenzi) [inline src/assets/scss/layouts/_forms.scss:59]: Prevent hardcoded media queries. To do that better, create a new partial, then declare device breakpoints to make them reusable. Maybe you can reach out to me to get help with this.

File: src/assets/scss/layouts/_header.scss [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    position: fixed;
+    top: 0;
+    left: 0;
+    z-index: 100;
+    background:rgba($dark,.5);
+     
+      &.shrink{
+        background:$white;
+        -webkit-box-shadow:0px 0px 8px $blue;
+        -moz-box-shadow:0px 0px 8px $blue;
```
Comment (eliemugenzi) [inline src/assets/scss/layouts/_header.scss]: Remove these `webkit` and `moz` declarations. CSS3 nolonger needs it.

File: src/assets/scss/layouts/_header.scss [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+       i.fa{
+         font-size: 27px;
+         color: $white;
+         // font-weight: 100;
+       }
+      }
+    }
+  }
+  
+  @media(min-width:768px){
```
Comment (eliemugenzi) [inline src/assets/scss/layouts/_header.scss]: Maybe I can show you a better way of using media queries in SASS

File: src/assets/scss/pages/_login.scss [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+                text-shadow:  0px 4px 4px rgba(0, 174, 255, 0.25);
+                letter-spacing: 0.02em;
+                text-align: center;
+            }
+            $base-spacing-unit: 22.5px;
+            $half-spacing-unit: 5px;
+            $border-radius-global: 15px;
+            $button-width: auto;
+            $button-height: 40px;
+            $button-text-color: white;
```
Comment (eliemugenzi) [inline src/assets/scss/pages/_login.scss]: Use `Hexadecimal` instead of `white`

File: src/assets/scss/main.scss [hunk 6]
```diff
@@ -1,6 +1,14 @@
-@import "./abstracts/variables";
-@import "./layouts/header";
-@import "./layouts/footer";
-@import "./components/home";
-@import "./pages/home";
-@import "./pages/login";
+@import './abstracts/variables';
```
Comment (eliemugenzi) [inline src/assets/scss/main.scss]: Rename this folder `abstracts` to `utils`. Prefer using this SASS file structure: [7-1 pattern structure](https://gist.github.com/rveitch/84cea9650092119527bc)

File: src/assets/scss/pages/_socialLogin.scss [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+  }
+  .btn-fb{
+    color: #fff;
+    background-color:#3b5998;
+  }
+  .checkbox{
+    color: white;
+  }
+  .btn-fb:hover{
+    color: #fff;
```
Comment (eliemugenzi) [inline src/assets/scss/pages/_socialLogin.scss]: Use variables

File: src/assets/scss/pages/_socialLogin.scss [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+    background-color:#496ebc 
+  }
+  .btn-tw{
+  color: #fff;
```
Comment (eliemugenzi) [inline src/assets/scss/pages/_socialLogin.scss]: Use variables instead

**Review by eliemugenzi — CHANGES_REQUESTED** [review]

File: src/assets/scss/pages/_socialLogin.scss (hunk 8 above)
Comment (EmyRukundo) [reply src/assets/scss/pages/_socialLogin.scss]: Thanks, I have corrected...

File: __tests__/actions/socialLogin.js (hunk 1 above)
Comment (EmyRukundo) [reply __tests__/actions/socialLogin.js]: Thanks @eliemugenzi, I have corrected...

File: src/assets/scss/pages/_socialLogin.scss (hunk 7 above)
Comment (EmyRukundo) [reply src/assets/scss/pages/_socialLogin.scss]: Thanks, I have corrected...

File: src/assets/scss/main.scss (hunk 6 above)
Comment (EmyRukundo) [reply src/assets/scss/main.scss]: Thanks, I have corrected...

**Review by UhiriweAudace — APPROVED** [review]

File: src/assets/scss/abstracts/_mobile.scss [hunk 9]
```diff
@@ -0,0 +1,27 @@
+//Small Devices
+@mixin mediaSm {
```
Comment (eliemugenzi) [inline src/assets/scss/abstracts/_mobile.scss]: Good job for this file. I recommend it to all of teammates to use it since they merge this PR.

**Review by eliemugenzi — APPROVED** [review]

**Pull request merged**